src/v/rpc/reconnect_transport.cc

```diff
@@ -34,6 +34,7 @@ static inline bool has_backoff_expired(
 }
 
 ss::future<> reconnect_transport::stop() {
+    _transport.shutdown();
```

Comment:
hmm... i don't follow this. what's the point in doing a _transport.stop() if you are doing a _transport.shutdown()

confusing to have 2 methods here.

Reply:
The `_transport.shutdown()` closes the connection underlying file descriptor. `stop()` method close the dispatch gate and fails all outstanding futures.

Reply:
why wouldn't stop close the underlying connection too of `_transport()` ? seems like the bug is right there no?

Reply:
we can include the `shutdown` call into `stop()`. if it was called before it would be a no op.

Reply:
actually this is not needed, I will drop this commit.